tree/ntuple/src/RFieldBase.cxx

```diff
@@ -554,49 +554,69 @@ ROOT::RFieldBase::Create(const std::string &fieldName, const std::string &typeNa
 
       if (!result) {
          auto cl = TClass::GetClass(typeName.c_str());
-         // NOTE: if the class is not at least "Interpreted" we currently don't try to construct
-         // the RClassField, as in that case we'd need to fetch the information from the StreamerInfo
-         // rather than from TClass. This might be desirable in the future, but for now in this
-         // situation we rely on field emulation instead.
-         if (cl != nullptr && cl->GetState() >= TClass::kInterpreted) {
+
+         if (cl != nullptr) {
```

Comment:
```suggestion
         if (cl != nullptr && cl->GetState() > TClass::kForwardDeclared) {
```
In the forward declared state, the TClass would only know there is a type and no information about it.